### DISCUSSION TITLE
Can renovate support automergeType for branch and PR together?

### DISCUSSION THREAD

**Discussion by ankitabhopatkar13** [renovatebot/renovate#11589] [Request Help] [answered]
We are currently using **automergeType: "branch"**, but at times when the pipelines fail, renovate creates a PR as expected.

But when renovate runs again as per the schedule and branch is rebased with the latest dependencies and if the pipeline passes, can we automerge the PR too?

**Comment by rarkins** [comment, marked as answer]
I'm pretty sure the automerge should still happen even if a build problem initially caused the branch automerge to fail. I know it *used* to, but I haven't used the logic myself in a while. If you have such a branch/PR still open you should be able to get a hint from the logs about whether it's trying automerge of the PR or not.

**Reply by ankitabhopatkar13** [reply]
Thanks for a very fast response @rarkins.

Looking at the logs, it indeed tries to see if the PR can be merged but gets a 406 response code.

```
"message": "Response code 406 (Not Acceptable)",
         "stack": "HTTPError: Response code 406 (Not Acceptable)\n    at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
```

**Reply by ankitabhopatkar13** [reply]
Ah, that is probably because our MR's need one approval before it can be merged. I see, I will try to see if I can make an exception for renovate branches and try again.

**Reply by ankitabhopatkar13** [reply]
So there is an option to ignore approvals for Gitlab as mentioned https://docs.renovatebot.com/configuration-options/#gitlabignoreapprovals

But that also explicitly says that the option only works when the below options are enabled, but in my case -

`automerge=true` - is enabled
`automergeType=pr` - this is set to `branch` but I believe this shouldn't be a problem
`gitLabAutomerge=true` - this is not set in our case because the PR's are not created everytime but only when pipeline fails.

Maybe I am missing something but I am not able to understand why is the third option a mandatory setting🤔

**Reply by rarkins** [reply]
Check the discussions from when that feature was created and you can hopefully see why that was (I forget). It may have been "conservative" and happy to have a feature request where you propose loosening the logic.
